redfish-core/lib/power_subsystem.hpp

```diff
@@ -25,6 +25,47 @@ namespace redfish
 inline void getPowerSubsystemAllocation(
     const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
+    // Get max from CapLimits
+    sdbusplus::asio::getAllProperties(
+        *crow::connections::systemBus, "org.open_power.OCC.Control",
+        "/xyz/openbmc_project/control/host0/power_cap_limits",
+        "xyz.openbmc_project.Control.Power.CapLimits",
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::DBusPropertiesMap& propertiesList) {
+            if (ec)
+            {
+                if (ec.value() != EBADR)
+                {
+                    BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
+
+            const uint32_t* minCap = nullptr;
+            const uint32_t* maxPowerCapValue = nullptr;
+            const bool success = sdbusplus::unpackPropertiesNoThrow(
+                dbus_utils::UnpackErrorPrinter(), propertiesList,
+                "MinPowerCapValue", minCap, "MaxPowerCapValue",
+                maxPowerCapValue);
+            if (!success)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            // If MaxPowerCapValue valid, store Allocation properties in JSON
+            if ((maxPowerCapValue != nullptr) &&
+                (*maxPowerCapValue < UINT32_MAX))
+            {
+                asyncResp->res.jsonValue["Allocation"]["AllocatedWatts"] =
+                    *maxPowerCapValue;
```

Comment:
If the following getAllProperties() below fails we will be returning these results now where we used to not return them since we only had a single call.

For the "AllocatedWatts" entry this could be an incorrect value if we actually do have PowerCapEnable on. From [Redfish schema](https://redfish.dmtf.org/schemas/v1/PowerSubsystem.v1_1_2.json):
`This property shall represent the total power currently allocated or budgeted to this subsystem.`

Seems like we might need to chain these getProperty calls together and fill in the response in a single function.

Reply:
I tried to chain them, but was not able to get it working.

Reply:
Since that is an error case, we would be allocating the full amount which is safer (system will not be consuming more than that). Seems better to overreport power allocation vs not reporting anything. If it triggers customer warning that is probably fine, because there is a problem reading the current power cap / enable status.

Reply:
I'm okay with this for downstream. I'm sure for upstream we will have to figure out how to chain these together for it to get merged there.

I'd like @gtmills to weigh in on the partial reporting aspect.